File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        m, c = 0,0
        for elem in nums:
            if (c == 0):
                m = elem
                
            if elem == m:
                c += 1
            else:
                c -= 1
            

        return m
            
        

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int m = 0;
        int c = 0;
        
        for (int elem: nums){
            if (c==0){
                m = elem;}
            if (elem == m){
                c++;}
            else
            {c--;}
        }
        
        return m;
        
        
    }
}